source/daplink/interface/swd_host.c

```diff
@@ -111,12 +112,38 @@ uint8_t swd_init(void)
     //       and fixed.
     DAP_Setup();
     PORT_SWD_SETUP();
+    swd_enabled = true;
     return 1;
 }
 
 uint8_t swd_off(void)
 {
+    // Clear CSYSPWRUPREQ and CDBGPWRUPREQ if SWD had been enabled. This is to ensure we don't
+    // leave the target's debug logic powered up.
+    if (swd_enabled)  {
+
+        // Ensure CTRL/STAT register selected in DPBANKSEL
+        if (!swd_write_dp(DP_SELECT, 0)) {
+            return 0;
+        }
+        if (!swd_write_dp(DP_ABORT, 0x04)) {
+            return 0;
+        }
+
+        // Power down.
+        if (!swd_write_dp(DP_CTRL_STAT, CDBGRSTREQ)) {
+            return 0;
+        }
+
+        //The debugger requires reading from it before it succesfully resets itself
+        uint32_t temp = 0;
+        if (!swd_read_dp(DP_CTRL_STAT, &temp)) {
+            return 0;
+        }
+    }
+
```

Comment:
swd_off is just for shutting off the pins. This logic should be added to swd_uninit_debug().